packages/venia-concept/src/components/Checkout/form.js

```diff
@@ -1,329 +1,364 @@
-                    >
-                        {this.shippingMethodSummary}
-                    </Section>
-                    <Section label="TOTAL">
-                        <Price
-                            currencyCode={cart.totals.quote_currency_code}
-                            value={cart.totals.subtotal || 0}
-                        />
-                        <br />
-                        <span>{cart.details.items_qty} Items</span>
-                    </Section>
-                </div>
-                <div className={classes.footer}>
-                    <Button onClick={this.dismissCheckout}>Back to Cart</Button>
-                    <SubmitButton
-                        submitting={submitting}
-                        valid={ready}
-                        submitOrder={submitOrder}
-                    />
-                </div>
-            </Fragment>
-        );
-    }
+const isSubmitDisabled = (busy, valid) => busy || !valid;
+
+/**
+ * The EditableForm component renders the actual edit forms for the sections
+ * within the form.
+ */
+const EditableForm = props => {
+    const {
+        editing,
+        editOrder,
+        submitShippingAddress,
+        submitShippingMethod,
+        submitPaymentMethodAndBillingAddress,
+        submitting,
+        isAddressIncorrect,
+        incorrectAddressMessage,
+        directory: { countries }
+    } = props;
+
+    const handleCancel = useCallback(() => {
+        editOrder(null);
+    }, [editOrder]);
+
+    const handleSubmitAddressForm = useCallback(
+        formValues => {
+            submitShippingAddress({
+                formValues
+            });
+        },
+        [submitShippingAddress]
+    );
+
+    const handleSubmitPaymentsForm = useCallback(
+        formValues => {
+            submitPaymentMethodAndBillingAddress({
+                formValues
+            });
+        },
+        [submitPaymentMethodAndBillingAddress]
+    );
+
+    const handleSubmitShippingForm = useCallback(
+        formValues => {
+            submitShippingMethod({
+                formValues
+            });
+        },
+        [submitShippingMethod]
+    );
+
+    switch (editing) {
+        case 'address': {
+            const { shippingAddress } = props;
 
-    get paymentMethodSummary() {
-        const { classes, hasPaymentMethod, paymentData } = this.props;
-
-        if (!hasPaymentMethod) {
             return (
-                <span className={classes.informationPrompt}>
-                    Add Billing Information
-                </span>
+                <AddressForm
+                    initialValues={shippingAddress}
+                    submitting={submitting}
+                    countries={countries}
+                    cancel={handleCancel}
+                    submit={handleSubmitAddressForm}
+                    isAddressIncorrect={isAddressIncorrect}
+                    incorrectAddressMessage={incorrectAddressMessage}
+                />
```

Comment:
This wouldn't block a 👍 but just pointing out that alphabetizing really helps readability in cases where there are a lot of props like this.

Reply:
Definitely. I did it in some places but not others. I shall look for and alphabetize the props on component implementations :)

Reply:
I think I'm tired of staring at this stuff. I did it where I could. If we want to enforce this we should add some linter/plugin that alphabetizes this for us automatically. Ugh. lol